crates/wasm-encoder/src/core/types.rs

```diff
@@ -611,6 +625,13 @@ impl<'a> CoreTypeEncoder<'a> {
         }
     }
 
+    fn encode_cont(&mut self, ty: &ContType) {
+        self.bytes.push(0x5d);
+        // Note that this is encoded as a signed type rather than unsigned
+        // as it's decoded as an s33
+        i64::from(ty.0).encode(self.bytes);
```

Comment:
I think, I copy-pasted it from my downstream implementation, where I think I copy pasted it from how `HeapType::Concrete` is encoded. Actually, I think I modelled a lot of `Cont` on how `Concrete` works for heap types. But maybe it fine to encode and decode continuation types as `u32`s?

Reply:
Actually, I retract the above. I looked at the reference implementation. There we encode it as a `s33`. But we have written in the explainer/informal spec that it is supposed to be encoded as a `typeidx`, which I believe per the spec should be encoded as an `u32`. So, there seems to be an inconsistency here. I will chat with @rossberg about it.

Reply:
I believe `s33` is used in binary encoding locations where a positive value indicates "this is a type index" and a negative value indicates "this is actually a construct which isn't a type index but it's a structure to parse". It's needed to make the binary representation more compact by not needing a prefix byte for differentiating between the two. In the context of `Cont`, though, I don't think it's necessary because the prefix byte of `Cont`, 0x5d, has already been pushed so there's no ambiguity of what the next item is. That's what makes me think that this may want to be a u32 instead of an s33, but I wanted to confirm to be sure

Reply:
I think you are correct. I spoke with Andreas about this. We decided to go with the `s33` encoding as it may be more forwards compatible as it gives us some extension point for adding future attributes to continuation types (should we ever want that). So I have proposed this change on the spec repo https://github.com/WebAssembly/stack-switching/pull/89.